### PR TITLE
Bump dev version

### PR DESCRIPTION
**Pull request by soapy1**
ref: https://github.com/conda-incubator/conda-store/issues/967

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ jobs:
       - name: "Build and check package - ${{ matrix.directory }} 📦"
         uses: hynek/build-and-inspect-python-package@v2
         id: baipp
+        permissions:
+          id-token: write # required for attest-build-provenance-github
```
Comment (peytondmurray) [inline .github/workflows/release.yaml:47]: I think we don't need this, as it is set on L20 above.

**Review by peytondmurray — CHANGES_REQUESTED** [review]
Small change requested!

File: .github/workflows/release.yaml (hunk 1 above)
Comment (soapy1) [reply .github/workflows/release.yaml:47]: without this addition we get this error https://github.com/conda-incubator/conda-store/actions/runs/11843467867/job/33004483291

**Comment by peytondmurray** [comment]
Merging as it seems the permissions set here don't apply to forks.

**Review by peytondmurray — APPROVED** [review]

**Pull request merged**